.markdownlint.yaml

```diff
@@ -0,0 +1,196 @@
+MD014: false
+
+# MD018/no-missing-space-atx - No space after hash on atx style heading
+MD018: true
+
+# MD019/no-multiple-space-atx - Multiple spaces after hash on atx style heading
+MD019: true
+
+# MD022/blanks-around-headings/blanks-around-headers - Headings should be surrounded by blank lines
+MD022:
+  # Blank lines above heading
+  lines_above: 1
+  # Blank lines below heading
+  lines_below: 1
+
+# MD023/heading-start-left/header-start-left - Headings must start at the beginning of the line
+MD023: true
+
+# MD025/single-title/single-h1 - Multiple top-level headings in the same document
+# TODO: consider enabling it
+
+# MD026/no-trailing-punctuation - Trailing punctuation in heading
+MD026:
+  # Punctuation characters
+  punctuation: ".,;:!。，；：！"
+
+# MD027/no-multiple-space-blockquote - Multiple spaces after blockquote symbol
+MD027: true
+
+# MD028/no-blanks-blockquote - Blank line inside blockquote
+MD028: true
+
+# MD029/ol-prefix - Ordered list item prefix
+MD029:
+  # List style
+  style: "one_or_ordered"
+
+# MD030/list-marker-space - Spaces after list markers
+MD030:
+  # Spaces for single-line unordered list items
+  ul_single: 1
+  # Spaces for single-line ordered list items
+  ol_single: 1
+  # Spaces for multi-line unordered list items
+  ul_multi: 1
+  # Spaces for multi-line ordered list items
+  ol_multi: 1
+
+# MD031/blanks-around-fences - Fenced code blocks should be surrounded by blank lines
+MD031:
+  # Include list items
+  list_items: true
+
+# MD033/no-inline-html - Inline HTML
+# MD033:
+#   # Allowed elements
+#   allowed_elements: ["br", "center", "img", "script", "form", "input"]
+
+# MD034/no-bare-urls - Bare URL used
+MD034: true
+
+# MD035/hr-style - Horizontal rule style
+MD035:
+  # Horizontal rule style
+  style: "---"
+
+# MD037/no-space-in-emphasis - Spaces inside emphasis markers
+MD037: true
+
+# MD038/no-space-in-code - Spaces inside code span elements
+MD038: true
+
+# MD039/no-space-in-links - Spaces inside link text
+MD039: true
+
+# MD040/fenced-code-language - Fenced code blocks should have a language specified
+MD040: true
+
+# MD041/first-line-heading/first-line-h1 - First line in a file should be a top-level heading
+MD041:
+  # Heading level
+  level: 1
+  # RegExp for matching title in front matter
+  front_matter_title: "^\\s*title\\s*[:=]"
+
+# MD042/no-empty-links - No empty links
+MD042: true
+
+# MD046/code-block-style - Code block style
+MD046:
+  # Block style
+  style: "fenced"
+
+# MD047/single-trailing-newline - Files should end with a single newline character
+MD047: true
+
+# MD048/code-fence-style - Code fence style
+MD048:
+  # Code fence style
+  style: "backtick"
+
+# MD049/emphasis-style - Emphasis style should be consistent
+MD049:
+  # Emphasis style should be consistent
+  style: "underscore"
+
+# MD050/strong-style - Strong style should be consistent
+MD050:
+  # Strong style should be consistent
+  style: "asterisk"
```

Comment:
Can we add [MD053](https://github.com/DavidAnson/markdownlint/blob/main/doc/md053.md)? Would it catch errors like the one fixed by https://github.com/TrenchBoot/TrenchBoot.github.io/pull/46/commits/bc3ffcff801aaafd24f1da5cd457ac8338cce9df?

Reply:
Yes:

```
markdownlint.............................................................Failed
- hook id: markdownlint
- exit code: 1

docs/blueprints/index.md:80:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Duplicate link or image reference definition: "linux"] [Context: "[linux]: https://lore.kernel.o..."]
```

After changing:

```diff
-| Linux     | [In Progress][linux-up]         | N/A
+| Linux     | [In Progress][linux]         | N/A
 | Xen       | [Partially In Progress][xen-ap] | [In Progress][xen-qos]
 
 [grub-up]: https://lists.gnu.org/archive/html/grub-devel/2024-08/msg00088.html
 [grub-qos]: https://github.com/QubesOS/qubes-grub2/pull/13
-[linux-up]: https://lore.kernel.org/lkml/20240826223835.3928819-1-ross.philipson@oracle.com/
+[linux]: https://lore.kernel.org/lkml/20240826223835.3928819-1-ross.philipson@oracle.com/
```

Added in https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/6959805f5ba51e7b7aeb1a826b4e58ff2fe821d1